src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts

```diff
@@ -119,4 +136,16 @@ export class DefaultResourcesNameStep extends AzureWizardPromptStep<DeployWorksp
             return undefined;
         });
     }
+
+    private async checkSuffixNamesAvailable(context: DeployWorkspaceProjectInternalContext, resourceName: string): Promise<boolean> {
```

Comment:
I think it makes more sense to set the names in the context or returning an object with the default names rather than returning a boolean. I'd expect the behavior to be that it would set the names that are available and only falling back to the resource name if it wasn't.

This current behavior is all or nothing.

Reply:
Yeah the all or nothing was intentional, you think partial would be okay?

Reply:
Personally thought it would be preferred to do partial, but if it was a design choice, I'll leave it to you.

Reply:
Changing it to allow partials!